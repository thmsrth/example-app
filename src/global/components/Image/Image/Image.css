.Image__wrapper {
  display: flex;
  flex-direction: column;
  margin-bottom: var(--section-margin);
}

.Image__container {
  display: flex;
  justify-content: space-between;
}

.Image__image {
  width: fit-content;
  height: fit-content;
}

.Image__delete {
  margin-right: var(--section-margin);
  margin-left: var(--section-margin);
  color: var(--color-action);
  background-color: var(--action-secondary-background-color);
  border: 0;
  padding: 0;
}

.Image__delete__icon {
  margin-right: 0;
}

.Button__icon.Image__delete__icon.ImageList__delete__icon.Button__icon--disabled.Image__delete__icon--disabled.ImageList__delete__icon--disabled {
  color: var(--action-disabled-background-color);
  background-color: var(--action-secondary-background-color);
  border: 0;
}
