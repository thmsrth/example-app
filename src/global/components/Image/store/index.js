import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';

import imageUpload from '../reducers/imageUpload';

const middleware = process.env.NODE_ENV === 'development' ? [thunk, createLogger({
  collapsed: (getState, action, logEntry) => !logEntry.error,
})] : [thunk];

const rootReducer = combineReducers({ imageUpload });

const store = createStore(rootReducer, applyMiddleware(...middleware));

export default store;
