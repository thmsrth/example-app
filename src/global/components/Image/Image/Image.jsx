import React from 'react';
import PropTypes from 'prop-types';

import { bemCls } from '../../../utils/ClassNameHelpers';
import I18nGlobal from '../../../utils/I18nGlobal';

import Button from '../../Button';

import './Image.css';

const Image = ({ image, className }) => {
  const {
    path,
    name,
    label,
    deleteOptions,
  } = image;

  const {
    showDelete,
    deleteAction,
    deleteCondition,
  } = deleteOptions || {};

  return (
    <div className={bemCls(`Image ${className}`, '__wrapper')}>
      {label
      && (
        <label className={bemCls(`Image ${className}`, '__label')}>
          {label}
        </label>
      )}
      <div className={bemCls(`Image ${className}`, '__container')}>
        {path
          ? <img src={path} alt={name} className={bemCls(`Image ${className}`, '__image')} />
          : I18nGlobal.t('en.image.states.noImage')
        }
        {showDelete
        && (
          <Button
            tooltip={I18nGlobal.t('en.actions.delete')}
            iconName="trash"
            className={bemCls(`Image ${className}`, '__delete')}
            onClick={deleteAction}
            disabled={deleteCondition !== undefined && !deleteCondition}
          />
        )}
      </div>
    </div>
  );
};

export default Image;

Image.propTypes = {
  image: PropTypes.shape({
    path: PropTypes.string,
    label: PropTypes.string,
    name: PropTypes.string,
    deleteOptions: PropTypes.shape({
      showDelete: PropTypes.bool,
      deleteAction: PropTypes.func,
      deleteCondition: PropTypes.bool,
    }),
  }),
  className: PropTypes.string,
};

Image.defaultProps = {
  image: {
    path: '',
    label: '',
    name: '',
    deleteOptions: {
      showDelete: false,
      deleteAction: () => {},
      deleteCondition: false,
    },
  },
  className: '',
};
