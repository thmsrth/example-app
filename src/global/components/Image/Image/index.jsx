import React from 'react';
import Image from './Image';

const ImageWrapper = props => <Image {...props} />;

export default ImageWrapper;
