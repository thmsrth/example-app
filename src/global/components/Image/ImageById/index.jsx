import React from 'react';
import { connect, Provider } from 'react-redux';

import store from '../store/index';

import {
  fetchImageById,
} from '../actions/imageUpload';

import {
  getImageLoading,
  getImage,
} from '../reducers/imageUpload';

import ImageByIdWrapper from './ImageByIdWrapper';

const mapStateToProps = (state, props) => ({
  loading: getImageLoading(state),
  image: getImage(state, props.value),
});

const mapDispatchToProps = dispatch => ({
  fetchImageById: params => dispatch(fetchImageById(params)),
});

const ConnectedContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ImageByIdWrapper);

const InputByIdContainer = props => (
  <Provider store={store}>
    <ConnectedContainer {...props} />
  </Provider>
);

export default InputByIdContainer;
