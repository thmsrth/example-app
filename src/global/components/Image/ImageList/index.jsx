import React from 'react';
import { connect, Provider } from 'react-redux';

import store from '../store/index';

import {
  fetchImagesByIds,
  resetImageUploadError,
} from '../actions/imageUpload';

import {
  getImageLoading,
  getImageUploadError,
  getImage,
  getUploadedImages,
} from '../reducers/imageUpload';

import ImageListWrapper from './ImageListWrapper';

const mapStateToProps = (state, props) => ({
  uploadedImages: getUploadedImages(state),
  loading: getImageLoading(state),
  error: getImageUploadError(state),
  image: getImage(state, props.value),
});

const mapDispatchToProps = dispatch => ({
  fetchImagesByIds: params => dispatch(fetchImagesByIds(params)),
  resetImageUploadError: () => dispatch(resetImageUploadError()),
});

const ConnectedContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(ImageListWrapper);

const ImageListContainer = props => (
  <Provider store={store}>
    <ConnectedContainer {...props} />
  </Provider>
);

export default ImageListContainer;
