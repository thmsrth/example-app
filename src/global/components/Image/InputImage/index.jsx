import React from 'react';
import { connect, Provider } from 'react-redux';

import store from '../store/index';

import {
  uploadImage,
  fetchImageById,
  resetImageUploadError,
} from '../actions/imageUpload';

import {
  getUploadedImages,
  getImageLoading,
  getImageUploading,
  getImageUploadError,
  getImage,
} from '../reducers/imageUpload';

import InputImageWrapper from './InputImageWrapper';

const mapStateToProps = (state, props) => ({
  uploadedImages: getUploadedImages(state),
  uploading: getImageUploading(state, props.name),
  loading: getImageLoading(state),
  uploadError: getImageUploadError(state),
  image: getImage(state, props.value),
});

const mapDispatchToProps = dispatch => ({
  uploadImage: params => dispatch(uploadImage(params)),
  fetchImageById: params => dispatch(fetchImageById(params)),
  resetImageUploadError: () => dispatch(resetImageUploadError()),
});

const ConnectedContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(InputImageWrapper);

const InputImageContainer = props => (
  <Provider store={store}>
    <ConnectedContainer {...props} />
  </Provider>
);

export default InputImageContainer;
