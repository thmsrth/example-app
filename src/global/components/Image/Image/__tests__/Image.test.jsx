import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Image from '../Image';

const props = {
  className: 'MyImage',
  image: {
    name: 'image',
    path: '/folder/img.jpg',
    label: 'Image',
  },
};

describe('Image', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<Image {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should render the component correctly with no image', () => {
    const tree = shallow(<Image {...props} image={{ label: 'Label with no image' }} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
