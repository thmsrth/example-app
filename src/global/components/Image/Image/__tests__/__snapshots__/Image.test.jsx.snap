// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render the component correctly 1`] = `
<div
  className="Image__wrapper MyImage__wrapper"
>
  <label
    className="Image__label MyImage__label"
  >
    Image
  </label>
  <div
    className="Image__container MyImage__container"
  >
    <img
      alt="image"
      className="Image__image MyImage__image"
      src="/folder/img.jpg"
    />
  </div>
</div>
`;

exports[`Image should render the component correctly with no image 1`] = `
<div
  className="Image__wrapper MyImage__wrapper"
>
  <label
    className="Image__label MyImage__label"
  >
    Label with no image
  </label>
  <div
    className="Image__container MyImage__container"
  >
    No image
  </div>
</div>
`;
